src/components/Tag/Tag.tsx

```diff
@@ -30,21 +45,36 @@ export function Tag({children, disabled = false, onRemove}: TagProps) {
     newDesignLanguage && styles.newDesignLanguage,
   );
 
+  const clickableProps = {
+    'aria-disabled': (onClick && disabled) || undefined,
+    tabIndex: !disabled && onClick ? 0 : undefined,
+    onClick: !disabled ? onClick : undefined,
+    onKeyPress: !disabled ? onClick : undefined,
+  };
```

Comment:
We can 🔥this now since `button` elements handle tabindex, keyboard a11y, and disabled interaction states under the hood.

```suggestion
```